workflow/rules/quant.smk

```diff
@@ -1,28 +1,141 @@
-rule kallisto_index:
+rule kallisto_cds_index:
     input:
-        "resources/transcriptome.cdna.fasta",
+        "resources/transcriptome_clean.cdna.fasta"
+        if config["experiment"]["3-prime-rna-seq"]["activate"]
+        else "resources/transcriptome.cdna.fasta",                
     output:
-        "results/kallisto/transcripts.idx",
+        "results/kallisto_cds/transcripts.idx",
     log:
-        "results/logs/kallisto/index.log",
+        "results/logs/kallisto_cds/index.log",
     conda:
         "../envs/kallisto.yaml"
     shell:
         "kallisto index -i {output} {input} 2> {log}"
-
-
-rule kallisto_quant:
+    
+rule kallisto_cds_quant:
     input:
         fq=get_trimmed,
-        idx="results/kallisto/transcripts.idx",
+        idx="results/kallisto_cds/transcripts.idx",
     output:
-        directory("results/kallisto/{sample}-{unit}"),
+        kallisto_folder=directory("results/kallisto_cds/{sample}-{unit}"),
+        bam_file="results/kallisto_cds/{sample}-{unit}/pseudoalignments.bam",
     log:
-        "results/logs/kallisto/quant/{sample}-{unit}.log",
+        "results/logs/kallisto_cds/quant/{sample}-{unit}.log",
     params:
         extra=kallisto_params,
     conda:
         "../envs/kallisto.yaml"
     shell:
-        "kallisto quant -i {input.idx} -o {output} "
-        "{params.extra} {input.fq} 2> {log}"
+        "kallisto quant -i {input.idx} -o {output.kallisto_folder} {params.extra} {input.fq} 2> {log}"
+
+if config["experiment"]["3-prime-rna-seq"]["activate"]:
+    
+    rule kallisto_3prime_index:
+        input:
+            "resources/transcriptome_canonical.3prime.fasta",
+        output:
+            "results/kallisto_3prime/transcripts.idx",
+        log:
+            "results/logs/kallisto_3prime/index.log",
+        conda:
+            "../envs/kallisto.yaml"
+        shell:
+            "kallisto index -i {output} {input} 2> {log}"
+
+    rule kallisto_3prime_quant:
+        input:
+            fq="results/canonical_reads/{sample}-{unit}.fastq",
+            idx="results/kallisto_3prime/transcripts.idx",
+        output:
+            kallisto_folder=directory("results/kallisto_3prime/{sample}-{unit}"),
+        log:
+            "results/logs/kallisto_3prime/quant/{sample}-{unit}.log",
+        params:
+            extra=kallisto_params,
+        conda:
+            "../envs/kallisto.yaml"
+        shell:
+            "kallisto quant -i {input.idx} -o {output.kallisto_folder} {params.extra} {input.fq} 2> {log}"
+
+if config["experiment"]["3-prime-rna-seq"]["activate"]:
```

Comment:
Above, there is the same if already. You can remove this one.
```suggestion
```

Reply:
Noted.